include/deal.II/fe/fe_poly_tensor.h

```diff
@@ -245,11 +248,33 @@ class FE_PolyTensor : public FiniteElement<dim, spacedim>
     // initialize fields only if really
     // necessary. otherwise, don't
     // allocate memory
+
+    bool update_transformed_shape_values = false;
+    bool update_transformed_shape_grads = false;
+    bool update_transformed_shape_hessian_tensors = false;
+
+    for (unsigned int i = 0; i < this->mapping_type.size(); ++i)
+      {
+        MappingType mapping_type = this->mapping_type[i];
+
+        if (mapping_type != mapping_none)
+          update_transformed_shape_values = true;
+
+        if ((mapping_type == mapping_raviart_thomas) ||
+            (mapping_type == mapping_piola) ||
+            (mapping_type == mapping_nedelec) ||
+            (mapping_type == mapping_contravariant))
+          update_transformed_shape_grads = true;
+
+        if (mapping_type != mapping_none)
+          update_transformed_shape_hessian_tensors = true;
```

Comment:
This is the exact same condition as you had above for `update_transformed_shape_values`, so you will get the exact same answer. Is this on purpose?

Reply:
Ah it was not on purpose. Seeing the original command looked like 

```
            if (mapping_type != mapping_none)
              data->untransformed_shape_hessian_tensors.resize(n_q_points);
```

I can just absorb that boolean as being equal to the `update_transformed_shape_values` boolean since both only check if the mapping is not `mapping_none`.